### PR TITLE
feature(api): jwt requests authentication and restriction access to resources

### PR DESCRIPTION
**Pull request by WizBhoo**
Hello Rui,
Could you please review this feature ? Thanks in advance.

Purpose : added JWT authentication to the API and define restriction access to ressources depending on the Client who sends the request.
Existing tests were adapted in consequences and i have added some other tests to check 403 response.

Link to my project issues : #11 / #14

### CODE REVIEW

File: tests/Functional/Controller/Api/ClientUserControllerTest.php [hunk 1]
```diff
@@ -79,14 +79,53 @@ protected function setUp(): void
...
+            ->get('lexik_jwt_authentication.encoder')
+            ->encode($data);
+
+        $this->client->request(
+            'GET',
+            self::USERS_LIST_URI,
+            [],
+            [],
+            ['HTTP_AUTHORIZATION' => 'Bearer '. $token]
+        );
```
Comment (ruiteix) [inline tests/Functional/Controller/Api/ClientUserControllerTest.php:93]: You can factorize this in a trait or base class method with the name `requestAuthenticated(string $login, string $verb, string $url, array $data = null)` ?

Comment (ruiteix) [reply tests/Functional/Controller/Api/ClientUserControllerTest.php:93]: Or you can create a function which the authorization header.

File: .env [hunk 2]
```diff
@@ -31,3 +31,9 @@ APP_SECRET=bdcc8a6b0149786fab4d6d366bb8325d
 # IMPORTANT: You MUST configure your server version, either here or in config/packages/doctrine.yaml
 DATABASE_URL=mysql://db_user:db_password@127.0.0.1:3306/db_name?serverVersion=5.7
 ###< doctrine/doctrine-bundle ###
+
+###> lexik/jwt-authentication-bundle ###
+JWT_SECRET_KEY=%kernel.project_dir%/config/jwt/private.pem
```
[the remark this answers is not in the document]
Comment (WizBhoo) [reply .env:36]: Yes you are right and it will be the purpose of my on-going issue #15

File: config/packages/nelmio_api_doc.yaml [hunk 3]
```diff
@@ -6,6 +6,14 @@ nelmio_api_doc:
             title: BileMo - API REST
             description: An hightech smartphone selection !
             version: 1.0.0
+        securityDefinitions:
+            Bearer:
+                type: apiKey
+                description: 'Value: Bearer {JWT}'
```
Comment (ruiteix) [inline config/packages/nelmio_api_doc.yaml:12]: :1st_place_medal:

**Pull request merged**